lib/ddtrace/contrib/extensions.rb

```diff
@@ -60,11 +67,23 @@ def configure(target = configuration, opts = {})
         module Settings
           InvalidIntegrationError = Class.new(StandardError)
 
-          def self.included(base)
-            # Add the additional options to the global configuration settings
-            base.instance_eval do
-              option :registry, default: Registry.new
-            end
+          # The registry only holds declarative constant values and cannot be modified.
+          # This option is a no-op and will be removed in the future.
+          #
+          # @deprecated Use `Datadog::Contrib::REGISTRY` instead
+          def registry
+            Datadog.logger.warn('Deprecated access to `Datadog.configuration.registry`, use `Datadog.registry` instead.')
+            Contrib::REGISTRY
```

Comment:
I updated the message.

The reason to use `Datadog::Contrib::REGISTRY` internally in the tracer is that it's a direct access to the constant, and it's the only safe way to access the registry before the tracer initializes.

`Datadog.registry` is only available after the tracer has fully initialized.

Today this makes no different because calls to `Datadog.registry` lazily trigger the initialization of the whole tracer, but after we eager initialize the tracer that won't be the case. Users will always receive the tracer initialized, but internal components might see it uninitialized during gem loading.